File: OOP/Aula_07-Herenca/Conta.java

public class Conta {

	protected double saldo;

	// Getters
	public double getSaldo()			{ return this.saldo; }

	// Setters
	public void setSaldo(double saldo_p)		{ this.saldo = saldo_p; }


	public void depositar (double deposito) {

		this.setSaldo ( this.getSaldo() + deposito );
	}

	public double sacar (double valor) {

		if (this.getSaldo() - valor >= 0) {

			this.setSaldo ( this.getSaldo() - valor );
			return valor;
		}
		else {

			System.out.println("Saldo insuficiente para efetuar a operacao");
			return 0;
		}

	}

	public void atualizar (double taxa) {

		this.setSaldo ( this.getSaldo() + (this.getSaldo() * taxa)  );
	}

}

File: OOP/Aula_08_Classes_Abstratas/Att_01-05/Entities/Conta.cs
namespace Entities
{
	public abstract class Conta
	{
		protected double saldo;
		public double Saldo
		{
			get { return saldo; }
			set { saldo = value; }
		}

		public abstract void Depositar(double deposito);
		public abstract void Atualizar(double taxa);
		public double Sacar(double valor)
		{
			if (saldo - valor >= 0)
			{
				saldo -= valor;
				return valor;
			}
			Console.WriteLine("Saldo insuficiente para efetuar a operação");
			return 0;
		}
	}
}
